hyperparameter_search.py

```diff
@@ -17,7 +18,7 @@
 from pytorchtools import EarlyStopping
 import warnings
 import logging
-from utils.model_utils import AVAILABLE_MODELS
+from utils.model_utils import AVAILABLE_MODELS, create_model_id_and_timestamp, save_dataset_of_torch_model
 from dataloaders.dataset_stats import get_normalization_mean_std
 from dataloaders.dataset_labels import get_dataset_labels
 
```

Comment:
Related to the patience suggestion in row 302 and 349

```suggestion
import math
```